src/site/filters/liquid.js

```diff
@@ -21,6 +21,40 @@ module.exports = function registerFilters() {
   liquid.filters.humanizeTimestamp = dt =>
     moment.unix(dt).format('MMMM D, YYYY');
 
+  liquid.filters.formatMobileTableData = entity => {
+    const labels = entity.fieldTable.value[0];
+    const rows = Object.values(entity.fieldTable.value).reduce(
+      (rows, row, rowIndex) => {
+        // If it's the header row, continue.
+        if (rowIndex === 0 || !row) return rows;
+
+        // Row can either be a string or an array of strings depending on cardinality.
+        if (typeof row === 'string') {
+          rows.push({ dataPoints: [{ label: labels[0], value: row }] });
+          return rows;
+        }
+
+        // Derive the key-value pairs (label + value).
+        const dataPoints = row.map((column, columnIndex) => ({
+          label: labels[columnIndex],
+          value: column,
+        }));
+
+        // Add the item to our items.
+        rows.push({ dataPoints });
+
+        // Continue.
+        return rows;
+      },
+      [],
+    );
+
+    return {
+      caption: entity.fieldTable.caption,
+      rows,
+    };
+  };
```

Comment:
You could slice off the first row of `rows` and include that as a separate field, `columnHeaders`. This way you could reduce the number of `if (first row)` during iterations. ->

```
return {
   tableHeader: rows[0],
  caption: entity.fieldTable.caption,
  rows: rows.slice(1)
}
```

There are more elegant ways to do it than my example though.